### PR TITLE
feat: Update Karpenter to `v0.23.0` and add `awsInterruptionQueueName` to ArgoCD GitOps config

### LINKED ISSUE #1334: feat: Allow to configure Karpenter with new parameters via ArgoCD
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

The new Karpenter example shows how to install last version of Karpenter which comes with an SQS queue to handle Spot termination signals.

I would like a way to allow me to send additional parameters to the ArgoCD helm chart to introcude new parameters without the need to update the blueprint module each time.

### PR DESCRIPTION
**Pull request by allamand**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Allow to provides dynamic configuration parameters from Terraform to the Karpenter helm chart.

This is used here to provide the SQS queue Arn:

```
  enable_karpenter                     = true
  karpenter_helm_config = {
    awsInterruptionQueueName = module.karpenter.queue_arn
    awsDefaultInstanceProfile = "${local.name}-${local.node_group_name}"
  }
  karpenter_node_iam_instance_profile        = module.karpenter.instance_profile_name
  karpenter_enable_spot_termination_handling = true
  karpenter_sqs_queue_arn  = module.karpenter.queue_arn  
```
### Motivation

- Resolves #1334

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by allamand** [comment]
This is linked to the add-on repo PR :https://github.com/aws-samples/eks-blueprints-add-ons/pull/133

File: modules/kubernetes-addons/karpenter/locals.tf [hunk 1]
```diff
@@ -43,10 +43,13 @@ locals {
...
-  argocd_gitops_config = {
+  argocd_gitops_config = merge(
+    {
     enable                    = true
     serviceAccountName        = local.service_account
     controllerClusterEndpoint = var.addon_context.aws_eks_cluster_endpoint
     awsDefaultInstanceProfile = var.node_iam_instance_profile
-  }
+  },
+  var.helm_config
```
Comment (csantanapr) [inline modules/kubernetes-addons/karpenter/locals.tf:53]: No other addon does this, I mean taking data from `var.helm_config` which currently is only when terraform does does the helm install via the helm_release module.

My impression was that parameters that are not absolutely required from terraform, the user will have the chance to  set in git repo and not go back to terraform when changing something about the addon.


@bryantbiggs @apoo Do we want to this `merge` with `var.helm_config` with all the addosn?

Comment (allamand) [reply modules/kubernetes-addons/karpenter/locals.tf:53]: This already exists also in the External-dns addon: https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/external-dns/main.tf#L5-L10.

Here it is used to pass additional value created by Terraform (here the SNS queue arn) to GitOps. the helm_config allow this to be dynamic without needed to manually add each field

Comment (csantanapr) [reply modules/kubernetes-addons/karpenter/locals.tf:53]: Hi @bryantbiggs I will delegate to you if we want to allow the injection of `var.helm_config` into gitops for one addon.

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
If values are required for the GitOps/ArgoCD deployment option, then they are required here as well. The required schemas should match between the GitOps/ArgoCD and Terraform options; adding an argument for the queue ARN makes sense here, but not the entire `var.helm_config`

**Comment by allamand** [comment]
@bryantbiggs i've remove the helm_config

**Review by bryantbiggs — APPROVED** [review]

**Comment by csantanapr** [comment]
Thank you @allamand for removing the whole helm_config usage

**Pull request merged**